pkg/resources/version_test.go

```diff
@@ -0,0 +1,55 @@
+package resources
+
+import "testing"
+
+func TestVersion(t *testing.T) {
+	scenarios := []struct {
+		Name            string
+		TestVersion     string
+		ExpectedVersion *Version
+		Verifier        func(version *Version, err error, t *testing.T)
+	}{
+		{
+			Name:        "test valid version",
+			TestVersion: "1.2.3",
```

Comment:
prob could improve on these test cases

Reply:
there's not a lot of complexity in that version class, these tests cover a good and a bad version string. What other tests do you imagine would be helpful?

Reply:
something with a ```-``` in it as it looks to be the case we expect that type of version?

Reply:
something with more than 3 points in its release ```1.2.2.2323123```

Reply:
added a bunch of tests now